conbench/entities/benchmark_result.py

```diff
@@ -35,7 +35,20 @@ class BenchmarkResult(Base, EntityMixin):
     case_id: Mapped[str] = NotNull(s.String(50), s.ForeignKey("case.id"))
     info_id: Mapped[str] = NotNull(s.String(50), s.ForeignKey("info.id"))
     context_id: Mapped[str] = NotNull(s.String(50), s.ForeignKey("context.id"))
+
+    # Follow
+    # https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#one-to-many
+    # There is a one-to-many relationship between Run (one) and BenchmarkResult
+    # (0, 1, many). Result is child. Run is parent. Child has column with
+    # foreign key, pointing to parent.
     run_id: Mapped[str] = NotNull(s.Text, s.ForeignKey("run.id"))
+    # Note that `immediate` means that " items should be loaded as the parents
+    # are loaded, using a separate SELECT statement, or identity map fetch for
+    # simple many-to-one references." Do this until we see that this is a perf
+    # problem. Use case: count results per run (that use case can be solved by
+    # adding a cache/counter into the DB, also see
+    # https://stackoverflow.com/q/45407402/145400)
+    run: Mapped[Run] = relationship("Run", lazy="immediate", back_populates="results")
```

Comment:
Yes, this will be interesting to measure. Let's be sure to deploy to the Arrow Conbench instance soon after merging this PR, monitor it, and revert if it isn't working well.

Reply:
Separately: if this works out to not degrade performance, we could probably refactor certain queries in this codebase that search for "other runs that have any of this run's case/context pairs" (e.g. in [get_baseline_run()](https://github.com/conbench/conbench/blob/409537e0ffa9ec2abb1e40b942d5cf7f6cb02f4d/conbench/entities/run.py#L223) and [set_z_scores()](https://github.com/conbench/conbench/blob/409537e0ffa9ec2abb1e40b942d5cf7f6cb02f4d/conbench/entities/history.py#L368))

Reply:
Another thing: did you try to generate a DB migration? Everything you touched in this PR seems to be confined to SQLAlchemy sugar/instructions, but I suppose there's a chance it touches the DB schema in some way.

Reply:
> Another thing: did you try to generate a DB migration?

Yes, tried! Was empty, right!